File: 251004/Snopko/src/main/java/com/example/restservice/dto/request/UserRequestTo.java
package com.example.restservice.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class UserRequestTo {
    @NotBlank(message = "Login cannot be blank")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String login;

    @NotBlank(message = "Password cannot be blank")
    @Size(min = 8, max = 128, message = "Password must be between 8 and 128 characters")
    private String password;

    @Size(min = 2, max = 64, message = "First name must be less than 64 characters")
    private String firstname;

    @Size(min = 2, max = 64, message = "Last name must be less than 64 characters")
    private String lastname;

    private Long id;

}

File: 251003/Kukhatskavolets/microblog/src/main/kotlin/by/kukhatskavolets/dto/requests/UserRequestTo.kt
package by.kukhatskavolets.dto.requests

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UserRequestTo(
    val id: Long = 0,

    @field:NotBlank(message = "Login may not be blank")
    @field:Size(min = 2, max = 64, message = "Login should be between 2 and 64 symbols")
    val login: String = "",

    @field:NotBlank(message = "Password may not be blank")
    @field:Size(min = 8, max = 128, message = "Password should be between 8 and 128 symbols")
    val password: String = "",

    @field:NotBlank(message = "Firstname may not be blank")
    @field:Size(min = 2, max = 64, message = "Firstname should be between 2 and 64 symbols")
    val firstname: String = "",

    @field:NotBlank(message = "Lastname may not be blank")
    @field:Size(min = 2, max = 64, message = "Lastname should be between 2 and 64 symbols")
    val lastname: String = ""
)